metaserver/contracts.go

```diff
@@ -75,13 +77,64 @@ func (server *MetaServer) postContractHandler() http.HandlerFunc {
 			return
 		}
 
-		// BUG(kincaid): DB will throw error if file already exists. Might want to check explicitly.
+		// TEST CODE! Make every contract cost 100 bucks for testing.
+		// contract.StorageFee = 100 * 1000
+		// MORE TEST CODE! Make the duration of the contract 1 week.
+		// contract.EndDate = time.Now().Add(time.Hour * 24 * 7)
+
+		// Make sure the renter has enough money to pay for the contract.
+		if contract.StorageFee > renter.Balance {
+			writeErr("cannot afford contract", http.StatusBadRequest, w)
+			return
+		}
+
+		// Subtract the balance of the contract from the renter's account.
+		// TODO: Add atomic DB operations to increment and decrement renter balances.
+		renter.Balance -= contract.StorageFee
+		err = server.db.UpdateRenter(renter)
+		if err != nil {
+			writeAndLogInternalError(err, w, server.logger)
+			return
+		}
+
+		startTime := time.Now()
```

Comment:
It may be best to leave LastPaymentTime unset here (the 0 value for time.Time). The payment info then better reflects what has actually happened (no payments have been made to the provider). 

Another issue with the current approach is that, for the first payment, the fraction of the elapsed time on the contract will be calculated based on the time elapsed since the contract was submitted, not since it's start date. These two could differ a lot, and it's the time since the start date that reflects what should actually be paid.

In our case, though, I think these issues have negligible/no impact, and I'm ok setting them aside.

Reply:
If we had more time, I think discussing our trust model and possible ways to try to fool the payment system by messing with the date would be good. The main reason I'm setting the start date on the server is so renters can't try to get out of paying a contract by messing with the dates. But, now that I think about it, the provider should be able to figure that out and cancel the contract.

Anyway, I think setting this aside is fine.